editor/components/block-list/with-hover-areas.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, findDOMNode } from '@wordpress/element';
+
+const withHoverAreas = ( WrappedComponent ) => {
+	class WithHoverAreasComponent extends Component {
+		constructor() {
+			super( ...arguments );
+			this.state = {
+				hoverArea: null,
+			};
+			this.onMouseLeave = this.onMouseLeave.bind( this );
+			this.onMouseMove = this.onMouseMove.bind( this );
+		}
+
+		componentDidMount() {
+			// Disable reason: We use findDOMNode to avoid unnecessary extra dom Nodes
```

Comment:
Was this the main reason for using `mousemove` instead of some floating invisible elements which trigger on `mouseenter`?

Reply:
No, the main reason is that `mouseenter` doesn't trigger on areas with lower z-index, since these areas would need a lower z-index to allow writing in the editables etc...